### PR TITLE
failing if go build/vet do not succeed

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

**Review by tadelesh — COMMENTED** [review]
Maybe `go mod tidy` and `go vet` also need to handle execution error just to be safe.

**Comment by seankane-msft** [comment]
> Maybe `go mod tidy` and `go vet` also need to handle execution error just to be safe.

I'll add to `go mod tidy` but `go vet` is essentially the same as our linter and the linter allows for `//nolint` so I think it's better to default to the linter.

File: eng/scripts/build.ps1 [hunk 1]
```diff
@@ -10,6 +10,7 @@ function Process-Sdk () {
     if ($clean) {
         Write-Host "##[command]Executing go clean -v ./... in " $currentDirectory
         go clean -v ./...
+        if ($LASTEXITCODE) {exit $LASTEXITCODE}
```
Comment (weshaggard) [inline eng/scripts/build.ps1:13]: nit:
```suggestion
        if ($LASTEXITCODE) { exit $LASTEXITCODE }
```

**Review by weshaggard — APPROVED** [review]

**Review by tadelesh — APPROVED** [review]

**Pull request merged**